src/vs/platform/quickinput/browser/commandsQuickAccess.ts

```diff
@@ -338,8 +344,6 @@ export class CommandsHistory extends Disposable {
 
 		if (CommandsHistory.cache && CommandsHistory.cache.limit !== this.configuredCommandsHistoryLength) {
 			CommandsHistory.cache.limit = this.configuredCommandsHistoryLength;
```

Comment:
Either we have to set `isDirty` here or trigger a save to ensure we element count is reduced in storage.

Reply:
Done